src/pudl/workflow/epacems.py

```diff
@@ -63,13 +63,15 @@ def write_epacems_parquet_files(df: pd.DataFrame, table_name: str, partition: Ep
     # once we update our dependency.
     if prefect.context.pudl_upload_to_gcs:
         output_path = os.path.join(
-            prefect.context.pudl_upload_to_gcs, "parquet", "epacems")
+            prefect.context.pudl_upload_to_gcs, prefect.context.pudl_run_id,
+            "parquet", "epacems")
     else:
         output_path = os.path.join(
             prefect.context.pudl_settings["parquet_dir"], "epacems")
     parquet.write_to_dataset(
         table,
         root_path=output_path,
+        filesystem=prefect.context.pudl_filesystem,
```

Comment:
I don't like this `fsspec` api but what I used in variety of other places is this approach:

```
fs, _, _ = fsspec.get_fs_token_path(pudl_upload_to_gcs)
```

To get the filesystem associated with given path. That may be better than to use the if/else on the path-prefixes (because, as it turns out, both `gs://` and `gcs://` are valid)

Reply:
I'm thinking that for clarity it might make sense to fetch the fsspec filesystem instance within `write_epacems_parquet_files` rather than passing it through `prefect.context` as the latter feels like a fragile code.

Even though we do use prefect.context for similar purposes, the design pattern where we instantiate a piece of data somewhere far from where it will eventually be used can easily result in an accidental breakage because this code dependency is not explicit.